fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -790,21 +736,20 @@ private void removeOrModifyTransactionAssociatedWithLoanChargeIfDueAtDisbursemen
             LoanTransaction transactionToRemove = null;
             List<LoanTransaction> transactions = getLoanTransactions() ;
             for (final LoanTransaction transaction : transactions) {
-                if (transaction.isRepaymentAtDisbursement() && transaction.getLoanChargesPaid().contains(loanCharge)) {
```

Comment:
why remove this??

Reply:
Type of loanCharge and transaction.getLoanChargePaid() are not compatible and so spotbugs throw an error. If they are incompatible, how is this going to ever be TRUE as the second expression in the if statement will be FALSE? I removed this so we can discuss with someone who has more domain knowledge and understands the code. I might be missing something here. @vishwasbabu  @nazeer1100126  @Nayan Help me understand this..please...

I may put this back and suppress the spotbug check for now.

Reply:
I am putting this back and suppressing SpotBugs for this method. We can take a look at it later.